File: CMPS3500_MiniScheme_Starter_Package/procedural/file_loader_stub.cpp
/*
  / NAME: Walter Morales (provided file loader stub)    /
  / NAME: Henry Arinaga, Alberto Molina, Peter Uzuriaga  /
  / ASGT: CHECKPOINT 1                                  /
  / ORGN: CSUB - CMPS 3500                              /
  / FILE: file_loader_stub.cpp                          /
  / DATE: 03/15/2026 provided by Walter Morales         /
  / DATE: 03/29/2026 added exception catch              /
*/

// Tiny file-loading helper stub for the procedural implementation.
// Replace or adapt as needed.
#include <fstream>
#include <sstream>
#include <string>

std::string read_file(const std::string& path) {
    std::ifstream in(path);
    //error handling: throw an exception if the file cannot be opened
    if (!in) {
        throw std::runtime_error("Could not open file: " + path);
    }
    std::ostringstream ss;
    ss << in.rdbuf();
    return ss.str();
}

File: CMPS3500_MiniScheme_Starter_Package/oop/FileLoaderStub.java
// Tiny file-loading helper stub for the OOP implementation.
import java.nio.file.Files;
import java.nio.file.Path;

public class FileLoaderStub {
    public static String readFile(String path) throws Exception {
        try {
        return Files.readString(Path.of(path));
        } catch (Exception e){
            throw new RuntimeException("Could not open file: " + path);
        }
    }
}
